dipy/denoise/enhancement_kernel.pyx

```diff
@@ -0,0 +1,420 @@
+import numpy as np
+cimport numpy as cnp
+cimport cython
+import os.path
+
+cimport safe_openmp as openmp
+from safe_openmp cimport have_openmp
+from cython.parallel import prange, threadid, parallel
+
+from dipy.data import get_sphere
+from dipy.core.sphere import disperse_charges, Sphere, HemiSphere
+from tempfile import gettempdir
+from libc.math cimport sqrt, exp, fabs, cos, sin, tan, acos, atan2
+from math import ceil
+
+cdef class EnhancementKernel:
+
+    cdef double D33
+    cdef double D44
+    cdef double t
+    cdef int kernelsize
+    cdef double kernelmax
+    cdef double [:, :] orientations
+    cdef double [:, :, :, :, ::1] lookuptable
+    cdef object sphere
+    cdef object num_threads
+
+    def __init__(self, D33, D44, t, force_recompute=False,
+                 orientations=None, test_mode=False, num_threads=None):
+        """ Compute a look-up table for the contextual
+        enhancement kernel
+
+        Parameters
+        ----------
+        D33 : float
+            Spatial diffusion
+        D44 : float
+            Angular diffusion
+        t : float
+            Diffusion time
+        force_recompute : boolean
+            Always compute the look-up table even if it is available
+            in cache. Default is False.
+        orientations : integer or Sphere object
+            Specify the number of orientations to be used with
+            electrostatic repulsion, or provide a Sphere object.
+            The default sphere is 'repulsion100'.
+        test_mode : boolean
+            Computes the lookup-table in one direction only
+            
```

Comment:
num_threads is not documented (though I have a sense for what it might be doing...). In particular, what happens in the default case?